lib/reporters/html/partials/scripts.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+exports.scripts = () => {
+
+    return `<script>
+  headings = [];
+
+  onload = function () {
+      headings = document.querySelectorAll('h2');
+      reset();
+  };
+
+  onscroll = function (e) {
+      var heading = find(window.scrollY);
+      if (!heading) {
+        var current = document.querySelector('#menu a.active');
+        if (current) {
+          current.className = '';
+        }
+        return;
+      }
+      var links = document.querySelectorAll('#menu a')
+              , link;
+
+      for (var i = 0, len = links.length; i < len; ++i) {
+          link = links[i];
+          link.className = link.getAttribute('href') == '#' + heading.id ? 'active' : '';
+      }
+  };
+
+  function find (y) {
+      var i = headings.length
+              , heading;
+
+      while (i--) {
+          heading = headings[i];
+          if (y >= heading.offsetTop) {
+              return heading;
+          }
+      }
+  }
+
+  function toggle (className) {
+
+      var elements = document.getElementsByClassName(className);
+
+      for (var i = 0, il = elements.length; i < il; ++i) {
```

Comment:
```suggestion
      for (var i = 0; i < elements.length; ++i) {
```
Same as before, I think this makes it a bit simpler.

Reply:
See comment on first instance of this.